worker-document/src/main/java/com/hpe/caf/worker/document/scripting/JavaScriptEngineLazy.java

```diff
@@ -31,7 +31,7 @@ public final class JavaScriptEngineLazy implements ObjectCodeProvider
 
     public JavaScriptEngineLazy()
     {
-        this.scriptEngine = CacheBuilder.newBuilder().concurrencyLevel(1).build(CacheLoader.from(JavaScriptEngineLazy::buildEngine));
+        this.scriptEngine = CacheBuilder.newBuilder().maximumSize(1).concurrencyLevel(1).build(CacheLoader.from(JavaScriptEngineLazy::buildEngine));
```

Comment:
I'm not 100% familiar with this class Caoimhe, so not sure if this is the right fix or not. I see there are `expireAfterAccess` and `expireAfterWrite` methods we could use to set an eviction policy as well, not sure if one of those might be better.

Maybe @dermot-hardy can take a look if he gets a chance.

Reply:
I will take a look into those methods in the meantime, thanks @rorytorneymf

Reply:
Hi @rorytorneymf, I think setting the maxsize would be a good approach for the following reasons;
At the moment the cache is set to never evict so it does not. I am guessing that this is why FOD flagged it. Setting an eviction policy does not seem the right approach as this cache is being used as a way to lazy load the ScriptEngine. The desired outcome is that this ScriptEngine remains in cache populated on startup of the worker by ApplicationImpl.
I would try checking in the maxsize and see if FOD is satisfied by the change, if its not then there's an argument to remain as is and over rule the FOD flag. We do not try to add additional alternative engines to the cache.